core/configuration/dt_config_provider.go

```diff
@@ -171,6 +173,15 @@ func validateConfiguration(config *DtConfiguration) error {
 }
 
 func generateAgentId() int64 {
-	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	var rng *rand.Rand
+
+	var seed [8]byte
+	_, err := cryptoRand.Read(seed[:])
+	if err != nil {
```

Comment:
Can we add a (debug) log in this case?

Reply:
This actually happens before we initialize our loggers, so it would be an "unconditional" log (which users could not configure), but yes, would be possible by simply logging something with `fmt.Println(...)` as we do at another point in the `DtConfigProvider` as well. Though I'm not sure this feels important enough to add logging for - what do you think?

Reply:
In that case, I don't think it's important enough in production, but make sure you verify at least manually that we don't always take this code path.